ext/oss/client.go

```diff
@@ -140,10 +145,20 @@ func (c *Client) GeneratePresignURL(destinationURI string, expirationInSeconds i
 	// set expiration time
 	expireAt := time.Now().Add(time.Duration(expirationInSeconds) * time.Second)
 	// generate presigned URL
-	presignedURL, err := c.Client.Presign(c.ctx, req, oss.PresignExpiration(expireAt))
+	presignResponse, err := c.Client.Presign(c.ctx, req, oss.PresignExpiration(expireAt))
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
-	return presignedURL.URL, nil
 
+	presignedURL := presignResponse.URL
+	parsedURL, err := url.Parse(presignedURL)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	if strings.Contains(parsedURL.Host, ossInternalEndpointSuffix) {
```

Comment:
can we just trim the `-internal` ? no need to do string checking

Reply:
updated. I use string.Replace to avoid confusing cases where the bucket contains `-internal`